todo.js

```diff
@@ -1 +1,208 @@
-// Write your code
\ No newline at end of file
+class ToDoApp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {allToDos: [], idCounter: 0};
+
+    this.createToDo = this.createToDo.bind(this);
+    this.updateToDo = this.updateToDo.bind(this);
+    this.editToDo = this.editToDo.bind(this);
+    this.deleteToDo = this.deleteToDo.bind(this);
+  }
+
+  componentDidMount() {
+    const allToDos = JSON.parse(localStorage.getItem("allToDos"));
+    const idCounter = JSON.parse(localStorage.getItem("idCounter"));
+
+    if (allToDos) {
+      this.setState({allToDos: allToDos, idCounter: idCounter});
+    }
+  }
+
+  createToDo(content) {
+    const todos = this.state.allToDos;
+    const id = this.state.idCounter + 1;
+    todos.push({id, content, isEditing: false});
+    this.setState({allToDos: todos, idCounter: id});
+
+    this.saveToDo(todos);
+    localStorage.setItem("idCounter", id);
+  }
+
+  updateToDo(id, content) {
+    const todos = this.state.allToDos.map((todo) => {
+      if (todo.id === id) {
+        todo.content = content;
+        todo.isEditing = false;
+      }
+      return todo;
+    });
+    this.setState({allToDos: todos});
+
+    this.saveToDo(todos);
+  }
+
+  deleteToDo(id) {
+    const todos = this.state.allToDos.filter((todo) => todo.id !== id);
+    this.setState({allToDos: todos});
+
+    this.saveToDo(todos);
+  }
+
+  editToDo(id) {
+    const todos = this.state.allToDos.map((todo) => {
+      if (todo.id === id) {
+        todo.isEditing = true;
+      }
+      return todo;
+    });
+    this.setState({allToDos: todos});
+
+    this.saveToDo(todos);
+  }
+
+  saveToDo(allToDos) {
```

Comment:
ToDo 全部を保存しているので、誤解のないように複数形にしておくといいかなーと思いました。

```suggestion
  saveToDos(allToDos) {
```

Reply:
メソッド名を複数形に修正しました。

https://github.com/Kassy0220/react-todo-app-practice/blob/ddc7d6d666d89d0e14f3aa70e836afc656960480/todo.js#L57